src/mono/mono/metadata/bundled-resources.c

```diff
@@ -0,0 +1,143 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+#include <glib.h>
+#include <stdbool.h>
+
+#include "bundled-resources-internals.h"
+#include "assembly-internals.h"
+
+#include <mono/metadata/assembly.h>
+#include <mono/metadata/mono-private-unstable.h>
+
+static GHashTable *bundled_resources = NULL;
+static bool bundle_contains_assemblies, bundle_contains_satellite_assemblies = false;
+
+//---------------------------------------------------------------------------------------
+//
+// mono_bundled_resources_free frees all memory allocated for bundled resources.
+// It should only be called when the runtime no longer needs access to the data,
+// most likely to happen during runtime shutdown.
+//
+
+void
+mono_bundled_resources_free (void)
+{
+	g_hash_table_destroy (bundled_resources);
+	bundled_resources = NULL;
+
+	bundle_contains_assemblies = false;
+	bundle_contains_satellite_assemblies = false;
+}
+
+//---------------------------------------------------------------------------------------
+//
+// mono_bundled_resources_add handles bundling of many types of resources to circumvent
+// needing to find or have those resources on disk. The MonoBundledResource struct models
+// the union of information carried by all supported types of resources which are
+// enumerated in MonoBundledResourceType.
+//
+// bundled_resources:
+// A single hash table will hold all resources being bundled with the understanding
+// that all resources being bundled are uniquely named. The bundled resource is tagged
+// with the type of resource it represents, and the pointer added to this hash table
+// should fully represent a MonoBundled*Resource struct defined in `bundled-resources-internals.h
+//
+// Arguments:
+// ** resources_to_bundle - An array of pointers to `MonoBundledResource`, which details
+//     the type of MonoBundled*Resource information follows this pointer in memory.
+// len - The number of resources being added to the hash table
+//
+
+void
+mono_bundled_resources_add (MonoBundledResource **resources_to_bundle, uint32_t len)
+{
+	if (!bundled_resources)
+		bundled_resources = g_hash_table_new (g_str_hash, g_str_equal);
+
+	gboolean assemblyAdded, satelliteAssemblyAdded;
```

Comment:
lets stick to one bool type in this source file. Since we already use bool as part of the header, then we should probably use bool through the implementation as well.

Reply:
Used bool, and removed all internal types (that I could recognize)